Block/Adminhtml/System/Config/Field/RecurringProcessingModel.php

```diff
@@ -16,25 +16,30 @@
 
 class RecurringProcessingModel extends Select
 {
-    /**
-     * Options
-     *
-     * @var array
-     */
-    protected $options = [
-        Vault::CARD_ON_FILE => 'Card on File',
-        Vault::UNSCHEDULED_CARD_ON_FILE => 'Unscheduled Card on File',
-        Vault::SUBSCRIPTION => 'Subscription'
-    ];
+    private function getSourceOptions(): array
+    {
+        return [
+            ['label' => 'Card on File', 'value' => Vault::CARD_ON_FILE],
+            ['label' => 'Unscheduled Card on File', 'value' => Vault::UNSCHEDULED_CARD_ON_FILE],
+            ['label' => 'Subscription', 'value' => Vault::SUBSCRIPTION]
+        ];
+    }
+
+    public function _toHtml(): string
+    {
+        if (!$this->getOptions()) {
+            $this->setOptions($this->getSourceOptions());
+        }
```

Comment:
Can we move that to the class constructor?

Reply:
I would rather keep it here because that class is an extension of `Magento\Framework\View\Element\Html` and this method will be called specifically. What do you think?

Reply:
I would favor move this to constructor for two reasons:

1. this method seems to be called many times, we can avoid the if condition
2. the options seems not changing during the execution, it's more sense to set it once in the constructor that we know it will be called once

Reply:
After a small discussion with @hossam-adyen, we decided to leave it as it is in order to follow Magento's extensibility best practices. An example implementation can be found in their [documentation](https://developer.adobe.com/commerce/php/tutorials/admin/create-dynamic-row-configuration).